pkg/util/utils_supported.go

```diff
@@ -74,19 +75,15 @@ func GetRootlessConfigHomeDir() (string, error) {
 	var rootlessConfigHomeDirError error
 
 	rootlessConfigHomeDirOnce.Do(func() {
-		cfgHomeDir := os.Getenv("XDG_CONFIG_HOME")
-		if cfgHomeDir == "" {
-			home := os.Getenv("HOME")
-			resolvedHome, err := filepath.EvalSymlinks(home)
-			if err != nil {
-				rootlessConfigHomeDirError = errors.Wrapf(err, "cannot resolve %s", home)
-				return
-			}
-			tmpDir := filepath.Join(resolvedHome, ".config")
-			st, err := os.Stat(tmpDir)
-			if err == nil && int(st.Sys().(*syscall.Stat_t).Uid) == os.Geteuid() && st.Mode().Perm() >= 0700 {
-				cfgHomeDir = tmpDir
-			}
+		cfgHomeDir, err := unshare.HomeDir()
```

Comment:
Shouldn't we use `c/storage/pkg/homedir` instead?

I'm a bit confused as I have not known about `unshare.HomeDir`. We usually use `homedir` in other places. Should we consolidate the unshare one into homedir?

Reply:
Also note that `unshare` does not look at XDG.

Reply:
Yes you are right, and I also found another bug in the code, where rootlessConfigHomeDirError need to be an external variable.